cmd/shellExecute.go

```diff
@@ -0,0 +1,134 @@
+package cmd
+
+import (
+	"net/url"
+	"os/exec"
+	"strings"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/pkg/errors"
+
+	"github.com/SAP/jenkins-library/pkg/command"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/SAP/jenkins-library/pkg/vault"
+)
+
+type shellExecuteUtils interface {
+	command.ExecRunner
+	FileExists(filename string) (bool, error)
+}
+
+type shellExecuteUtilsBundle struct {
+	*vault.Client
+	*command.Command
+	*piperutils.Files
+}
+
+func newShellExecuteUtils() shellExecuteUtils {
+	utils := shellExecuteUtilsBundle{
+		Command: &command.Command{},
+		Files:   &piperutils.Files{},
+	}
+	utils.Stdout(log.Writer())
+	utils.Stderr(log.Writer())
+	return &utils
+}
+
+func shellExecute(config shellExecuteOptions, telemetryData *telemetry.CustomData) {
+	utils := newShellExecuteUtils()
+
+	err := runShellExecute(&config, telemetryData, utils)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runShellExecute(config *shellExecuteOptions, telemetryData *telemetry.CustomData, utils shellExecuteUtils) error {
+	// create vault client
+	// try to retrieve existing credentials
+	// if it's impossible - will add it
+	vaultConfig := &vault.Config{
+		Config: &api.Config{
+			Address: config.VaultServerURL,
+		},
+		Namespace: config.VaultNamespace,
+	}
+	client, err := vault.NewClientWithAppRole(vaultConfig, GeneralConfig.VaultRoleID, GeneralConfig.VaultRoleSecretID)
```

Comment:
where is this used?
I don't see usage of `client`. Do I overlook something=

Doesn't this clash with the default Vault functionality?

Reply:
fixed
client variable already skipped, but need to call NewClientWithAppRole for resolving test credentials anyway